mirrord-layer/src/tcp_mirror.rs

```diff
@@ -0,0 +1,170 @@
+use std::{
+    borrow::Borrow,
+    collections::HashSet,
+    hash::{Hash, Hasher},
+};
+
+use anyhow::Result;
+use async_trait::async_trait;
+use mirrord_protocol::{ConnectionID, NewTCPConnection, TCPClose, TCPData};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::TcpStream,
+    select,
+    sync::mpsc::{channel, Receiver, Sender},
+    task,
+};
+use tokio_stream::{wrappers::ReceiverStream, StreamExt};
+use tracing::{debug, error};
+
+use crate::{common::Listen, tcp::TCPHandler};
+
+async fn tcp_tunnel(mut local_stream: TcpStream, remote_stream: Receiver<Vec<u8>>) {
+    let mut remote_stream = ReceiverStream::new(remote_stream);
+    let mut buffer = vec![0; 1024];
+    loop {
+        select! {
+            message = remote_stream.next() => {
+                match message {
+                    Some(data) => {
+                        match local_stream.write_all(&data).await {
+                            Ok(_) => {},
+                            Err(err) => {error!("writing to local stream err {err:?}"); break;}
+                        }
+                    },
+                    None => {
+                        debug!("tcp tunnel exiting due to remote stream closed");
+                        break;
+                    }
+                }
+            },
+            res = local_stream.read(&mut buffer) => {
```

Comment:
Add documentation, something like: "Read the application's response from the socket and discard the data, so that the socket doesn't fill up."